src/XMakeBuildEngine/Construction/ProjectRootElement.cs

```diff
@@ -1935,16 +1935,20 @@ private XmlDocumentWithLocation LoadDocument(string fullPath)
             {
                 try
                 {
-#if MSBUILDENABLEVSPROFILING 
+#if MSBUILDENABLEVSPROFILING
                     string beginProjectLoad = String.Format(CultureInfo.CurrentCulture, "Load Project {0} From File - Start", fullPath);
                     DataCollection.CommentMarkProfile(8806, beginProjectLoad);
 #endif
-                    using (XmlTextReader xtr = new XmlTextReader(fullPath))
+                    XmlReaderSettings dtdSettings = new XmlReaderSettings();
+                    dtdSettings.DtdProcessing = DtdProcessing.Ignore;
+
+                    using (var stream = new StreamReader(fullPath, true))
```

Comment:
I stumbled upon this [here](https://github.com/Microsoft/msbuild/blob/master/src/XMakeBuildEngine/UnitTestsPublicOM/Construction/ProjectRootElement_Tests.cs#L670) where encoding is added with BOM and we should be able to read this encoding instead of using a default one.